iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/BusAdapter.java

```diff
@@ -168,51 +168,77 @@ public double getQ() {
         return getDelegate().getQ();
     }
 
-    // -------------------------------
-    // Not implemented methods -------
-    // -------------------------------
     @Override
-    public int getConnectedTerminalCount() {
-        throw MergingView.createNotImplementedException();
+    public int getConnectedComponentNumber() {
+        return getDelegate().getConnectedComponentNumber();
     }
 
     @Override
-    public Iterable<Line> getLines() {
-        throw MergingView.createNotImplementedException();
+    public void setConnectedComponentNumber(int connectedComponentNumber) {
+        getDelegate().setConnectedComponentNumber(connectedComponentNumber);
     }
 
     @Override
-    public Stream<Line> getLineStream() {
-        throw MergingView.createNotImplementedException();
+    public void setSynchronousComponentNumber(int synchronousComponentNumber) {
+        getDelegate().setSynchronousComponentNumber(synchronousComponentNumber);
     }
 
     @Override
-    public Iterable<DanglingLine> getDanglingLines() {
-        throw MergingView.createNotImplementedException();
+    public int getSynchronousComponentNumber() {
+        return getDelegate().getSynchronousComponentNumber();
```

Comment:
I'm not sure it's working. I think we should not change the connected/synchronous component number of the underlying bus. If I access to the underlying network, the component numbers could be wrong, because there are valid only for a CGM point of view, not an IGM.

I think the BusAdapter should handle its own copy of the component numbers. Same remark for the TerminalAdapter